src/NuGet.Core/NuGet.Commands/RestoreCommand/Diagnostics/UnresolvedMessages.cs

```diff
@@ -45,20 +44,21 @@ public static async Task LogAsync(IEnumerable<IRestoreTargetGraph> graphs, Remot
                     messageTasks.Add(GetMessageAsync(
                         ddi.Framework.ToString(),
                         unresolved,
-                        remoteLibraryProviders, sourceCacheContext,
-                        logger,
+                        context.FilterDependencyProvidersForLibrary(unresolved),
```

Comment:
Applying package namespaces configuration to filter the sources if any `PackageDownload` item is unresolved during restore.